README.md

```diff
@@ -66,85 +66,118 @@ See [parse feature](features/parse.feature) for the details on using the `drafte
 
-The `drafter_parse_blueprint_to` takes a source blueprint and returns the given
-blueprint parsed and serialized as [API
-Elements](http://api-elements.readthedocs.io) in YAML or JSON.
+The `drafter_parse_blueprint_to` function translates a buffered API blueprint into [API
+Elements](http://api-elements.readthedocs.io), serialized into one of its supported serialization formats.
 
 ```c
-int drafter_parse_blueprint_to(const char* source, char ** out, const drafter_options options);
+drafter_error drafter_parse_blueprint_to(
+    const char* source,
+    char** out,
+    const drafter_parse_options* parse_opts,
+    const drafter_serialize_options* serialize_opts);
+);
 ```
 
-```c
-#include <drafter/drafter.h>
+Given a pointer to a UTF-8 encoded c-string,
 
+```c
 const char* blueprint =
   "# My API\n"
   "## GET /message\n"
   "+ Response 200 (text/plain)\n"
   "\n"
   "      Hello World!\n";
+```
 
-drafter_options options;
-options.format = DRAFTER_SERIALIZE_JSON;
-options.sourcemap = true;
+##### Serialized as YAML
 
-char* result = NULL;
-if (drafter_parse_blueprint_to(blueprint, &result, options) == 0) {
-    printf("%s\n", result);
-    free(result);
+Without options, the resulting API Elements is serialized as YAML.
+
+```c
+char* yamlApie = NULL;
+if (drafter_parse_blueprint_to(blueprint, &yamlApie, NULL, NULL) == 0) {
+    printf("%s\n", yamlApie);
 }
+
+free(yamlApie);
 ```
 
-#### Checking the validity of a blueprint
+##### Serialized as JSON
 
-The `drafter_check_blueprint` function allows checking the validity of a
-blueprint. This function will return a `drafter_result` when the blueprint
-produces warnings and/or errors.  With a `drafter_result`, the
-`drafter_serialize` function can be used to serialized the result as [API
-Elements](http://api-elements.readthedocs.io) in YAML or JSON.
+Tweaking `drafter_serialize_options` allows serialization into JSON.
 
+```c
+drafter_serialize_options* options = drafter_init_serialize_options();
+drafter_set_format(options, DRAFTER_SERIALIZE_JSON);
+
+char* jsonApie = NULL;
+if (0 == drafter_parse_blueprint_to(blueprint, &jsonApie, NULL, NULL)) {
+    printf("%s\n", jsonApie);
+}
+
+free(jsonApie);
+drafter_free_options(options);
+```
+
+#### Validate API Blueprint
+
+API Blueprint can be validated via `drafter_check_blueprint`. 
 
 ```c
-drafter_result* drafter_check_blueprint(const char* source);
+drafter_error drafter_check_blueprint(
+    const char* source,
+    drafter_result** res,
+    const drafter_parse_options* parse_opts);
 ```
 
+##### Simple validation
+
+The return value of `drafter_check_blueprint` indicates validation success.
+
 ```c
-#include <drafter/drafter.h>
+drafter_result* result = NULL;
 
-const char* blueprint =
-  "# My API\n"
-  "## GET /message\n"
-  "+ Response 200 (text/plain)\n"
-  "\n"
-  "      Hello World!\n";
+if (0 == drafter_check_blueprint(blueprint, result)) {
+    print("Understood.\n");
+} else {
+    print("Invalid API Blueprint.?\n");
```

Comment:
Checking `drafter_error`, does this function return an error result if the API Blueprint is invalid, or does it pass success with the parse result?

Looking at the code I can see that `DRAFTER_EINVALID_OUTPUT` is if I don't set `result` to a pointer which is more of a programatic error than the API Blueprint being invalid.

`DRAFTER_EINVALID_INPUT` is if I pass `NULL` source, and I'd imagine that `DRAFTER_EUNKNOWN` is perhaps unexpected. So none of these cases appear to be invalid/errenous API Blueprint.

Reply:
I usually think like this:
- Is the input of my interface a reference to a value?
  - If yes, can I use a reference (i.e. do I have C++)?
    - If yes, use a reference.
    - If not, **use a pointer and `assert` on it (first statement(s) of implementation)**.
  - If not, is the input of my interface a pointer to a value?
    - etc.

Reply:
I would remove the programming error values from `drafter_error` and just assert. The programmer wants an explosion, not a valid (error) value, on programatic errors, no?

What we are currently doing is basically lifting a programatic error into userland. And that's probably no business of libdrafter, which is nothing but a set of archives implementing text transformations.